lib/scanner/csv-scanner/csv-scanner.h

```diff
@@ -64,25 +64,28 @@ void csv_scanner_options_set_quotes(CSVScannerOptions *options, const gchar *quo
 void csv_scanner_options_set_quote_pairs(CSVScannerOptions *options, const gchar *quote_pairs);
 void csv_scanner_options_set_null_value(CSVScannerOptions *options, const gchar *null_value);
 
-typedef struct _CSVScanner
-{
-  CSVScannerOptions *options;
+typedef struct {
   GList *current_column;
   const gchar *src;
   GString *current_value;
   gchar current_quote;
+} CSVScannerIterator;
+
+typedef struct _CSVScanner
+{
+  CSVScannerOptions *options;
 } CSVScanner;
 
```

Comment:
The naming bothers me a little, CSVScanner was already implying some form of iteration. what's the role of the two data structures:

CSVScannerIterator is used to scan through the CSV formatted line, but what is CSVScanner?

If at all possible I'd stick to the CSVScanner name for the iterator role. Why can't we get rid off CSVScanner?